File: invesdwin-scripting-parent/invesdwin-scripting-julia/invesdwin-scripting-julia-runtime-contract/src/test/java/de/invesdwin/context/julia/runtime/contract/InputsAndResultsTestFloat.jl
println("getFloat")
if isdefined(Main, :getFloat) && !isnothing(getFloat)
	error("getFloat already defined!")
end
getFloat = putFloat
println(typeof(getFloat))
println(getFloat)
if typeof(getFloat) != Float32
	error("getFloat not Float32!")
end

println("getFloatVector")
if isdefined(Main, :getFloatVector) && !isnothing(getFloatVector)
	error("getFloatVector already defined!")
end
getFloatVector = putFloatVector
println(eltype(getFloatVector))
println(getFloatVector)
if eltype(getFloatVector) != Float32
	error("getFloatVector not Float32!")
end

println("getFloatVectorAsList")
if isdefined(Main, :getFloatVectorAsList) && !isnothing(getFloatVectorAsList)
	error("getFloatVectorAsList already defined!")
end
getFloatVectorAsList = putFloatVectorAsList
println(eltype(getFloatVectorAsList))
println(getFloatVectorAsList)
if eltype(getFloatVectorAsList) != Float32
	error("getFloatVectorAsList not Float32!")
end

println("getFloatMatrix")
if isdefined(Main, :getFloatMatrix) && !isnothing(getFloatMatrix)
	error("getFloatMatrix already defined!")
end
getFloatMatrix = putFloatMatrix
println(eltype(getFloatMatrix))
println(getFloatMatrix)
if eltype(getFloatMatrix) != Float32
	error("getFloatMatrix not Float32!")
end

println("getFloatMatrixAsList")
if isdefined(Main, :getFloatMatrixAsList) && !isnothing(getFloatMatrixAsList)
	error("getFloatMatrixAsList already defined!")
end
getFloatMatrixAsList = putFloatMatrixAsList
println(eltype(getFloatMatrixAsList))
println(getFloatMatrixAsList)
if eltype(getFloatMatrixAsList) != Float32
	error("getFloatMatrixAsList not Float32!")
end

File: invesdwin-scripting-parent/invesdwin-scripting-runtime-lua/src/test/java/de/invesdwin/scripting/runtime/lua/tests/InputsAndResultsTestFloat.lua
arraysToString = java.method(java.import('de.invesdwin.util.collections.Arrays'),'toString','java.lang.Object')

print("getFloat")
if(getFloat ~= nil) then
	error("getFloat already defined!")
end
getFloat = putFloat
getFloatType = type(getFloat)
print(getFloatType)
print(getFloat)
if(getFloatType ~= "number") then
	error("getFloat not Float!")
end

print("getFloatVector")
if(getFloatVector ~= nil) then
	error("getFloatVector already defined!")
end
getFloatVector = putFloatVector
getFloatVectorType = type(getFloatVector[1])
print(getFloatVectorType)
print(arraysToString(getFloatVector))
if(getFloatVectorType ~= "number") then
	error("getFloatVector not Float!")
end

print("getFloatVectorAsList")
if(getFloatVectorAsList ~= nil) then
	error("getFloatVectorAsList already defined!")
end
getFloatVectorAsList = putFloatVectorAsList
getFloatVectorAsListType = type(getFloatVectorAsList[1])
print(getFloatVectorAsListType)
print(arraysToString(getFloatVectorAsList))
if(getFloatVectorAsListType ~= "number") then
	error("getFloatVectorAsList not Float!")
end

print("getFloatMatrix")
if(getFloatMatrix ~= nil) then
	error("getFloatMatrix already defined!")
end
getFloatMatrix = putFloatMatrix
getFloatMatrixType = type(getFloatMatrix[1][1])
print(getFloatMatrixType)
print(arraysToString(getFloatMatrix))
if(getFloatMatrixType ~= "number") then
	error("getFloatMatrix not Float!")
end

print("getFloatMatrixAsList")
if(getFloatMatrixAsList ~= nil) then
	error("getFloatMatrixAsList already defined!")
end
getFloatMatrixAsList = putFloatMatrixAsList
getFloatMatrixAsListType = type(getFloatMatrixAsList[1][1])
print(getFloatMatrixAsListType)
print(arraysToString(getFloatMatrixAsList))
if(getFloatMatrixAsListType ~= "number") then
	error("getFloatMatrixAsList not Float!")
end
